src/coreclr/vm/mono/mono_coreclr.cpp

```diff
@@ -481,9 +481,33 @@ extern "C" EXPORT_API MonoClass* EXPORT_CC mono_class_from_mono_type(MonoType *i
 // Note: This function has NOT been generated by the source generator
 MonoClass * mono_class_from_name(MonoImage *image, const char* name_space, const char *name, bool ignoreCase)
 {
+    CONTRACTL
+    {
+        THROWS;
+        GC_TRIGGERS;
+        PRECONDITION(image != NULL);
+        PRECONDITION(name_space != NULL);
+        PRECONDITION(name != NULL);
+    }
+    CONTRACTL_END;
+
     GCX_PREEMP(); // temporary until we sort out our GC thread model
 
-    return g_HostStruct->class_from_name(image, name_space, name, ignoreCase);
+    MonoClass* monoClass = NULL;
+
+    EX_TRY
+    {
+        monoClass = g_HostStruct->class_from_name(image, name_space, name, ignoreCase);
```

Comment:
Last week I landed https://github.com/Unity-Technologies/runtime/pull/178 which means this is never going to throw.  If this is never going to throw, then I don't see how adding a try/catch here is going to do anything.

Reply:
@mrvoorhe I couldn't test on the latest as #178 is not yet in the unity repo (requires attribute methods changes as DO_API fail), if you say that try/catch is automatically added to `g_HostStruct->class_from_name` then there is no need in the PR :) 
I can see we have it here https://github.com/Unity-Technologies/runtime/pull/178/files#diff-4d3791b94ebacb3f48f99ad8b0cb63cd2a6bd48932f2ecb1122b9b355b6865aeR84, but I wasn't sure whether or not is covers `mono_class_from_name` as well as the comment says it is not generated

Reply:
I can see how the comment makes it harder to follow.  The native code is not generated.  However, there is still a C# method that is generated.

The changes here to `mono_class_from_name` can be dropped.  You can keep the other change in this PR if you'd like.

Reply:
I confirm that the #178 changes make exception to be added to the log.
it looks nice and clean like
```
TypeLoadException: Type 'UnityEngine.U2D.IK.IKManager2D' from assembly 'Unity.2D.IK.Runtime, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null' is attempting to implement an inaccessible interface.
   at System.Reflection.RuntimeAssembly.GetType(QCallAssembly assembly, String name, Boolean throwOnError, Boolean ignoreCase, ObjectHandleOnStack type, ObjectHandleOnStack keepAlive, ObjectHandleOnStack assemblyLoadContext) in C:\build\output\Unity-Technologies\runtime\src\coreclr\System.Private.CoreLib\Microsoft.Interop.LibraryImportGenerator\Microsoft.Interop.LibraryImportGenerator\LibraryImports.g.cs:line 1221
   at System.Reflection.RuntimeAssembly.GetType(String name, Boolean throwOnError, Boolean ignoreCase) in C:\build\output\Unity-Technologies\runtime\src\coreclr\System.Private.CoreLib\src\System\Reflection\RuntimeAssembly.cs:line 214
   at Unity.CoreCLRHelpers.CoreCLRHost.class_from_name(IntPtr image, SByte* name_space, SByte* name, Boolean ignoreCase) in C:\build\output\Unity-Technologies\runtime\unity\unity-embed-host\CoreCLRHost.cs:line 92
   at Unity.CoreCLRHelpers.CoreCLRHost.class_from_name_native(IntPtr image, SByte* name_space, SByte* name, Boolean ignoreCase) in C:\build\output\Unity-Technologies\runtime\unity\unity-embed-host\UnityEmbedHost.Generator\UnityEmbedHost.Generator.CallbacksGenerator\GeneratedCoreCLRHost.gen.cs:line 60
UnityEngine_CoreModule_ProxyFunctionPointers:UnityEngine_StackTraceUtility_ExtractStringFromExceptionInternal_UnmanagedCallers(IntPtr, ManagedSpanWrapper*, ManagedSpanWrapper*, IntPtr)
UnityEngine.DebugLogHandler:Internal_LogException(Exception, Object)
UnityEngine.DebugLogHandler:LogException(Exception, Object)
UnityEngine.Logger:LogException(Exception, Object)
UnityEngine.Debug:LogException(Exception)
UnityEngine.<>c:<RegisterUECatcher>b__0_0(Object, UnhandledExceptionEventArgs) (at C:\UnitySrc\neutron.git\Runtime\Export\Scripting\UnhandledExceptionHandler.bindings.cs:46)
System.Reflection.RuntimeAssembly:GetType(QCallAssembly, String, Boolean, Boolean, ObjectHandleOnStack, ObjectHandleOnStack, ObjectHandleOnStack) (at C:\build\output\Unity-Technologies\runtime\src\coreclr\System.Private.CoreLib\Microsoft.Interop.LibraryImportGenerator\Microsoft.Interop.LibraryImportGenerator\LibraryImports.g.cs:1221)
System.Reflection.RuntimeAssembly:GetType(String, Boolean, Boolean) (at C:\build\output\Unity-Technologies\runtime\src\coreclr\System.Private.CoreLib\src\System\Reflection\RuntimeAssembly.cs:214)
Unity.CoreCLRHelpers.CoreCLRHost:class_from_name(IntPtr, SByte*, SByte*, Boolean) (at C:\build\output\Unity-Technologies\runtime\unity\unity-embed-host\CoreCLRHost.cs:92)
Unity.CoreCLRHelpers.CoreCLRHost:class_from_name_native(IntPtr, SByte*, SByte*, Boolean) (at C:\build\output\Unity-Technologies\runtime\unity\unity-embed-host\UnityEmbedHost.Generator\UnityEmbedHost.Generator.CallbacksGenerator\GeneratedCoreCLRHost.gen.cs:60)

(Filename: C:/build/output/Unity-Technologies/runtime/unity/unity-embed-host/CoreCLRHost.cs Line: 92)
```

thanks @mrvoorhe ! 
I don't think we need to land this PR - will close and delete branch